src/v8/v8.cc

```diff
@@ -274,7 +279,7 @@ std::unique_ptr<WasmVm> V8::clone() {
   clone->store_ = wasm::Store::make(engine());
 
   clone->module_ = wasm::Module::obtain(clone->store_.get(), shared_module_.get());
-
+  clone->name_section = name_section;
```

Comment:
I'm not convinced that this is safe. Notably, `getCustomSection()` returns a `string_view` pointing to a part of `source_`, but that's owned by the main thread, and not workers, which means that by the time this is accessed during panic, the underlying data might not be there anymore (I'm not 100% sure if that's the case, but it seems dangerous to rely on that behavior). Perhaps you should build the function names index during `load()`?

Reply:
Actually, aren't those functions names available in `module_functions_` already?

Reply:
> Actually, aren't those functions names available in module_functions_ already?
It does not contain function index information, so we have to parse the name section anyway.

> Perhaps you should build the function names index during load()?
will follow this path

Reply:
https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/66/commits/351010923f4124eb79a5e8839156f949131eebb7 resolved and verified it's working